lib/Widgets/SeekBar.vala

```diff
@@ -16,34 +16,232 @@
  *
  * Granite.SeekBar will get the style class .app-notification
  *
- * {{../../doc/images/Toast.png}}
+ * {{../../doc/images/SeekBar.png}}
  */
 
 public class Granite.Widgets.SeekBar : Gtk.Grid {
+    /*
+     * The time of the full duration of the playback.
+     */
+    public double playback_duration { get; construct set; }
+
+    /*
+     * The progression of the playback as a decimal from 0.0 to 1.0.
+     */
+    public double playback_progress { get; private set; }
+
+    /*
+     * If the pointer is grabbing the scale button.
+     */
+    public bool is_grabbing { get; private set; }
+
+    /*
+     * If the pointer is hovering over the scale.
+     */
+    public bool is_hovering { get; private set; }
+
+
+    /*
+     * The left label that displays the time progressed.
+     */
+    public Gtk.Label progression_label { get; private set; }
+
+    /*
+     * The right label that displays the total duration time.
+     */
+    public Gtk.Label time_label {  get; private set;  }
+
+    /*
+     * The time of the full duration of the playback.
+     */
+    public Gtk.Scale scale {  get; private set;  }
 
-    /**
-     * Creates a new SeekBar
+
+    /*
+     * Emitted when the scale's window is hovered over.
+     */
+    public signal void scale_hover (Gdk.EventCrossing event);
+
+    /*
+     * Emitted when the pointer leaves the scale's window.
+     */
+    public signal void scale_leave (Gdk.EventCrossing event);
+
+    /*
+     * Emitted when the pointer moves over the scale.
+     */
+    public signal void scale_motion (Gdk.EventMotion event);
+
+    /*
+     * Emitted when the scale button is pressed.
      */
-    public SeekBar () {
+    public signal void scale_button_press (Gdk.EventButton event);
+
+    /*
+     * Emitted when the scale button is released.
+     */
+    public signal void scale_button_release (Gdk.EventButton event);
+
+    /*
+     * Emitted when a scroll action is performed on the scale.
+     */
+    public signal void scroll_action (Gtk.ScrollType scroll, double new_value);
+
+    /*
+     * Emitted when the region has been allocated for the scale.
+     */
+    //public signal void scale_size_allocate (Gtk.Allocation alloc_rect);
+
+    /*
+     * Creates a new SeekBar with a fixed playback duration.
+     * */
+    public SeekBar (double playback_duration) {
+        Object (playback_duration: playback_duration);
     }
 
     construct {
-        var progression_label = new Gtk.Label ("0:00");
+        get_style_context ().add_class ("seek-bar");
 
-        var scale = new Gtk.Scale.with_range (Gtk.Orientation.HORIZONTAL, 0, 1, 0.1);
-        scale.draw_value = false;
+        /* GUI */
+        orientation = Gtk.Orientation.HORIZONTAL;
+        column_spacing = 6;
+        halign = Gtk.Align.CENTER;
+        progression_label = new Gtk.Label ("");
+        progression_label.get_style_context ().add_class ("progression-label");
+        time_label = new Gtk.Label ("");
```

Comment:
Can this `time_label` go underneath the `progression_label` label initialization? It would make things clearer to see.